src/routes/article.py

```
@@ -89,87 +130,147 @@ class BodyUpdateArticle(BaseModel):
+    if current_user.role < board.writing_permission_level:
+        raise HTTPException(403, detail="You are not allowed to write to this board")
 
     article.title = body.title
     article.board_id = body.board_id
     article.updated_at = datetime.now(timezone.utc)
-    try: session.commit()
+    try:
+        session.commit()
     except IntegrityError:
         session.rollback()
         raise HTTPException(status_code=409, detail="unique field already exists")
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)